event/listener.php

```diff
@@ -37,19 +38,36 @@ public function __construct()
 	static public function getSubscribedEvents()
 	{
 		return array(
-			'core.text_formatter_s9e_render_after'	=> 'modify_img_bbcode',
+			'core.text_formatter_s9e_configure_after' => 'modify_img_template',
 		);
 	}
 
 	/**
-	* Modify image BBCode
+	* Modify image BBCode's template
 	*
 	* @param \phpbb\event\data $event The event object
 	* @return void
 	* @access public
 	*/
-	public function modify_img_bbcode($event)
+	public function modify_img_template($event)
 	{
-		$event['html'] = preg_replace("/<img src=\"(.*?)\" class=\"postimage\" alt=\"(.*?)\">/i", "<img data-original=\"$1\" class=\"postimage\" alt=\"$2\"><noscript><img src=\"$1\" class=\"postimage\" alt=\"$2\"></noscript>", $event['html']);
+		$tag   = $event['configurator']->tags['img'];
+		$dom   = $tag->template->asDOM();
+		$xpath = new DOMXPath($dom);
+		foreach ($xpath->query('//img[@src]') as $img)
```

Comment:
Shouldn't this be something like
`//img[@src and @class='postimage']`
to replace only BBCode [img] and maybe also attachment thumbnails?